blog/_posts/2019-09-05-put-down-particle-accelerator.md

```diff
@@ -0,0 +1,78 @@
+---
+slug: put-down-particle-accelerator
+date: 2019-09-05
+author: dan
+layout: blog
+title: "Microservice Devs: Put Down the Particle Accelerator"
+subtitle: "You Need a Chemistry Set"
+image: slac.jpg
+image_caption: "This is too much"
+tags:
+  - kubernetes
+  - microservices
+  - tilt
+keywords:
+  - tilt
+  - development
+---
+
+## The Right Tool for the Right Job
+Build tools don't work for microservices. We need a new kind of tool: the Microservice Development Engine (MDE). For decades, apps grew by building bigger binaries. So tools got better by getting faster at building big binaries. Now apps grow by adding more binaries ("microservice”), and each binary is smaller. Building each binary is easier, but requires more wrangling across binaries.
+
+Google, Facebook and others invested Engineer-Millenia in making build tools that build huge binaries quickly, such as [Bazel](https://bazel.build) and [Buck](https://buck.build). These became big science projects delivering amazing technical achievements, comparable to building a two-mile long particle accelerator.
+
+(Particle accelerators show up a few times in this post, so some background: atomic physics experiments smash atoms together to see what happens. The first particle accelerator had a 4-inch diameter and cost $25. Over time, physicists got more demanding. Today's best particle accelerator is [CERN](https://en.wikipedia.org/wiki/CERN): a 17 mile tunnel straddling the France-Switzerland border that cost $4.75 Billion to build.)
+
+For most chemists, a $15 chemistry set helps more than a multi-billion dollar particle accelerator. If you want to develop microservices, a tool built for that will be better than a build tool like Bazel.
+
+What does a Microservice Development Engine look like? It looks more like the build tools of 40 years ago: handling incremental updates to free you to focus on your app, not your build.
+
+## Why Build Tools
```

Comment:
idk how to fix this, but fwiw it took me a while to parse this as "Why 'Build Tools'" -- kept reading it as "why we should build tools"

Reply:
Thanks